client/client.go

```diff
@@ -2944,26 +2945,24 @@ DISCOLOOP:
 				continue
 			}
 
-			// Query the members from the region that Consul gave us, and
-			// extract the client-advertise RPC address from each member
-			var membersResp structs.ServerMembersResponse
-			if err := c.connPool.RPC(region, addr, "Status.Members", rpcargs, &membersResp); err != nil {
+			srv := &servers.Server{Addr: addr}
+			nomadServers = append(nomadServers, srv)
+
+			// Query the client-advertise RPC addresses from the region that
+			// Consul gave us
+			var members []string
+			if err := c.connPool.RPC(region, addr, "Status.RPCServers", rpcargs, &members); err != nil {
```

Comment:
If Nomad Servers are properly advertising their client-to-server IP address and port, then why do we have to call _any_ Nomad RPC to resolve server addresses? Aren't the addresses in Consul fine? If the addresses in Consul aren't fine, can we have Nomad Servers register a new Consul Service for Nomad Clients to discover without having to use an ancillary RPC?

Reply:
That's what I wondered too! But I dug _way_ back in the code and it looks like the reason it was done this way was that multiple regions could be registered in the same Consul cluster (?!). So we hit the RPC endpoint to say "ok but which of these are _our_ servers?" Although to be honest, this seems like a very strange topology that probably breaks a million other things.

Reply:
That being said, maybe we could tag the registration in Consul with the region name and then just query on that. That seems like it'd be nicer. Let me take a look at how much of a lift that'd be.

Reply:
Done in 094d3ba49675a36f57405ba7f497d3bf4cb0ea27 @schmichael. This seems like it cuts thru the extra nonsense here nicely. (And lets clients become live a smidgen faster!)